index.test-d.ts

```diff
@@ -133,3 +133,35 @@ convertElement()
 convertElement(null)
 convertElement(undefined)
 expectError(convertElement<Article>())
+
+declare const node: unknown
+
+/* Type assertion */
+if (isElement(node)) {
+  expectType<Element>(node)
+}
+
+if (isElement(node, (node): node is Section => node.tagName === 'section')) {
+  expectType<Section>(node)
+}
+
+/**
+ * Should prefer `isElement(node) && node.children.length > 0`,
+ * see below comments for details.
+ */
+if (isElement(node, (node) => node.children.length > 0)) {
+  expectType<unknown>(node)
+} else {
+  expectType<unknown>(node)
+}
+
+/**
+ * This is suggested to be used because `isElement(node)` is a correct assertion,
+ * but with a second param `test` which is not an assertion,
+ * the whole `isElement(node, test)` expression can not be an assertion anymore.
```

Comment:
```suggestion
 * This is the suggested use of this package so TypeScript can infer what types
 * it’s working with.
 * This way, `node` as an `Element` in the if-branch, and it could still be an
 * element (or something else) in the else-branch.
```